includes/class-wc-stripe-order-handler.php

```diff
@@ -82,6 +82,9 @@ public function process_redirect_payment( $order_id, $retry = true, $previous_er
 
 			WC_Stripe_Logger::log( "Info: (Redirect) Begin processing payment for order $order_id for the amount of {$order->get_total()}" );
 
+			// Lock the order.
+			$this->lock_order_payment( $order );
+
```

Comment:
I think we should surround this in the same condition as the webhook handler in case the webhook comes in first.

```
if ( $this->lock_order_payment( $order ) ) {
	return;
}
```

FWIW, when testing locally my webhook was arriving before this function.